pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -80,36 +83,106 @@ func (i *KumaInjector) InjectKuma(pod *kube_core.Pod) error {
 		if pod.Spec.InitContainers == nil {
 			pod.Spec.InitContainers = []kube_core.Container{}
 		}
-		pod.Spec.InitContainers = append(pod.Spec.InitContainers, i.NewInitContainer(pod))
+		ic, err := i.NewInitContainer(pod)
+		if err != nil {
+			return err
+		}
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, ic)
+	}
+
+	if err := i.reconcileProbes(pod); err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func needInject(pod *kube_core.Pod, ns *kube_core.Namespace) (bool, error) {
-	isEnabled := func(enabled string) (bool, error) {
-		switch enabled {
-		case metadata.KumaSidecarInjectionEnabled:
-			return true, nil
-		case metadata.KumaSidecarInjectionDisabled:
-			return false, nil
+func (i *KumaInjector) reconcileProbes(pod *kube_core.Pod) error {
```

Comment:
`reconcile` is as generic as `manager`, `data`, `service`. I read it as `somethingProbes`.

How about `overrideProbes` or `overrideHTTPProbes` or `overrideProbesWithVirtual`? something like this

Reply:
I agree, `overrideHTTPProbes` sounds better. I also considered `virtualizeProbes` but probably it also doesn't make things clearer